openedx/core/djangoapps/course_apps/rest_api/v1/views.py

```diff
@@ -59,9 +60,10 @@ class CourseAppSerializer(serializers.Serializer):  # pylint: disable=abstract-m
     def to_representation(self, instance: CourseApp) -> Dict:
         course_key = self.context.get("course_key")
         request = self.context.get("request")
+        app_status = self.context.get("app_status")
         data = {
             "id": instance.app_id,
-            "enabled": is_course_app_enabled(course_key, instance.app_id),
+            "enabled": app_status.get(instance.app_id, is_course_app_enabled(course_key, instance.app_id)),
```

Comment:
I think only 'is_course_app_enabled' would be enough here. And inside that method we can get the enabled_state either from the cache or from the actual db call (In this case also backfill the cache)

Reply:
I think that defeats the entire purpose of this PR since calling that function for a single app at a time will again cause it to have 8 queries for each course.